barretenberg/cpp/src/barretenberg/flavor/flavor.hpp

```diff
@@ -98,8 +98,35 @@ class PrecomputedEntitiesBase {
  * @tparam FF The scalar field on which we will encode our polynomial data. When instantiating, this may be extractable
  * from the other template paramter.
  */
+template <typename FF, typename CommitmentKey_> class ProvingKey_ {
+  public:
+    size_t circuit_size;
+    bool contains_recursive_proof;
+    std::vector<uint32_t> recursive_proof_public_input_indices;
+    bb::EvaluationDomain<FF> evaluation_domain;
+    std::shared_ptr<CommitmentKey_> commitment_key;
+    size_t num_public_inputs;
+    size_t log_circuit_size;
+
+    // Offset off the public inputs from the start of the execution trace
+    size_t pub_inputs_offset = 0;
+
+    // The number of public inputs has to be the same for all instances because they are
+    // folded element by element.
+    std::vector<FF> public_inputs;
+
+    ProvingKey_() = default;
+    ProvingKey_(const size_t circuit_size, const size_t num_public_inputs)
+    {
+        this->commitment_key = std::make_shared<CommitmentKey_>(circuit_size + 1);
+        this->evaluation_domain = bb::EvaluationDomain<FF>(circuit_size, circuit_size);
+        this->circuit_size = circuit_size;
+        this->log_circuit_size = numeric::get_msb(circuit_size);
+        this->num_public_inputs = num_public_inputs;
+    };
+};
 template <typename PrecomputedPolynomials, typename WitnessPolynomials, typename CommitmentKey_>
-class ProvingKey_ : public PrecomputedPolynomials, public WitnessPolynomials {
+class ProvingKeyAvm_ : public PrecomputedPolynomials, public WitnessPolynomials {
```

Comment:
oof, hate to invite non-uniformity into this base file... what's this about? can it be moved into AVM flavor?

Reply:
Hmm yeah I mean the only reason I had to keep this around is because it's hard to do this refactor for the AVM. The flavor is also generated so I can't just paste it in there without making work for the bonobos but could do I suppose